src/CabalHelper/Compiletime/CompPrograms.hs

```diff
@@ -92,8 +92,11 @@ patchBuildToolProgs SStack progs = do
 createProgSymlink :: FilePath -> FilePath -> IO ()
 createProgSymlink bindir target
   | [exe] <- splitPath target = do
-    Just exe_path <- findExecutable exe
-    createSymbolicLink exe_path (bindir </> takeFileName target)
-  | otherwise = do
-    cwd <- getCurrentDirectory
-    createSymbolicLink (cwd </> target) (bindir </> takeFileName target)
+    mb_exe_path <- findExecutable exe
```

Comment:
This could also be written as:
```suggestion
    | [exe] <- splitPath target
    , Just exe_path <- findExecutable exe = do
```
to avoid additional case expression

Reply:
Mmm, and remove Nothing, right?

Reply:
Maybe you cant use IO in guard patterns? Or i am missing something?

```haskell
createProgSymlink :: FilePath -> FilePath -> IO ()
createProgSymlink bindir target
  | [exe] <- splitPath target
  , Just exe_path <- findExecutable exe = do
      createSymbolicLink exe_path (bindir </> takeFileName target)
  | otherwise = createSymLinkFromCwd
  where createSymLinkFromCwd = do
          cwd <- getCurrentDirectory
          createSymbolicLink (cwd </> target) (bindir </> takeFileName target)
```
throws
```
* Couldn't match expected type `IO (Maybe FilePath)'
              with actual type `Maybe FilePath'
* In the pattern: Just exe_path
  In a stmt of a pattern guard for
                 an equation for `createProgSymlink':
    Just exe_path <- findExecutable exe
  In an equation for `createProgSymlink':
.....
```

Reply:
oh you are right. I am sorry for this wrong nitpick :/

Reply:
no problem and please, keep on it :smile_cat: